File: backend/src/main/java/br/com/academia/faceaccess/domain/Maquina.java
package br.com.academia.faceaccess.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.Instant;

@Entity
@Table(name = "maquina")
public class Maquina {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 120)
    private String nome;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private MaquinaTipo tipo;

    @Column(length = 80)
    private String marca;

    @Column(length = 80)
    private String modelo;

    @Column(name = "codigo_patrimonio", unique = true, length = 50)
    private String codigoPatrimonio;

    @Column(length = 120)
    private String localizacao;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private MaquinaStatus status = MaquinaStatus.ATIVA;

    @Column(length = 500)
    private String observacoes;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt = Instant.now();

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt = Instant.now();

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public MaquinaTipo getTipo() {
        return tipo;
    }

    public void setTipo(MaquinaTipo tipo) {
        this.tipo = tipo;
    }

    public String getMarca() {
        return marca;
    }

    public void setMarca(String marca) {
        this.marca = marca;
    }

    public String getModelo() {
        return modelo;
    }

    public void setModelo(String modelo) {
        this.modelo = modelo;
    }

    public String getCodigoPatrimonio() {
        return codigoPatrimonio;
    }

    public void setCodigoPatrimonio(String codigoPatrimonio) {
        this.codigoPatrimonio = codigoPatrimonio;
    }

    public String getLocalizacao() {
        return localizacao;
    }

    public void setLocalizacao(String localizacao) {
        this.localizacao = localizacao;
    }

    public MaquinaStatus getStatus() {
        return status;
    }

    public void setStatus(MaquinaStatus status) {
        this.status = status;
    }

    public String getObservacoes() {
        return observacoes;
    }

    public void setObservacoes(String observacoes) {
        this.observacoes = observacoes;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Instant createdAt) {
        this.createdAt = createdAt;
    }

    public Instant getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(Instant updatedAt) {
        this.updatedAt = updatedAt;
    }
}

File: frontend/src/types/maquina.ts
export type MaquinaTipo = 'CARDIO' | 'MUSCULACAO' | 'FUNCIONAL' | 'OUTRO';

export type MaquinaStatus = 'ATIVA' | 'MANUTENCAO' | 'INATIVA';

export interface MaquinaSummary {
  id: number;
  nome: string;
  tipo: MaquinaTipo;
  status: MaquinaStatus;
  localizacao: string | null;
  createdAt: string;
}

export interface Maquina {
  id: number;
  nome: string;
  tipo: MaquinaTipo;
  marca: string | null;
  modelo: string | null;
  codigoPatrimonio: string | null;
  localizacao: string | null;
  status: MaquinaStatus;
  observacoes: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateMaquinaRequest {
  nome: string;
  tipo: MaquinaTipo;
  marca?: string;
  modelo?: string;
  codigoPatrimonio?: string;
  localizacao?: string;
  observacoes?: string;
}

export interface UpdateMaquinaRequest {
  nome: string;
  tipo: MaquinaTipo;
  marca?: string;
  modelo?: string;
  codigoPatrimonio?: string;
  localizacao?: string;
  observacoes?: string;
}
